test/model/message_test.dart

```diff
@@ -1702,6 +1702,22 @@ void main() {
         check(store).messages.values
           .single.flags.deepEquals([MessageFlag.starred, MessageFlag.read]);
       });
+
+      test('avoid duplicate flags', () async {
+        await prepare();
+        final message = eg.streamMessage(flags: [MessageFlag.read]);
+        await prepareMessages([message]);
+
+        // Add 'read' flag again
+        await store.handleEvent(mkAddEvent(MessageFlag.read, [message.id]));
+
+        // Should verify no notifications since no actual change happened
+        // But current implementation might notify anyway if it doesn't check for change
+        // optimization. The main point is flags list shouldn't have duplicates.
+        // Let's check the flags first.
```

Comment:
I appreciate wanting to explain this 🙂 but I think we can cut this comment; the test is about an edge case that should be rare (in the fetch-event race) and an extra `notifyListeners` isn't a significant concern here.